### PR TITLE
CORE-9850 Docs for pre-auth tokens

### PR DESCRIPTION
**Pull request by charlieR3**

### CODE REVIEW

File: wiki/Manual-Approval.md [hunk 1]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
 ```
 
 </details>
+
+# Pre-authentication of registration requests
+
+The network operator can decide to pre-authenticate registering members, which will allow the registering member to by-pass any approval rules that have been defined for the group as described previously. Authentication is done outside of Corda using any criteria the network operator chooses. Once the network operator is finished their authentication process, they can generate a one-time-use pre-authentication token, also known as a pre-auth token, specific to the member that has been authenticated.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
The network operator can decide to pre-authenticate registering members, which will allow the registering member to by-pass any approval rules that have been defined for the group as described previously. Authentication is done outside of Corda using any criteria the network operator chooses. Once the network operator has finished their authentication process, they can generate a one-time-use pre-authentication token, also known as a pre-auth token, specific to the member that has been authenticated. 
```

File: wiki/Manual-Approval.md [hunk 2]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+
+<details>
+<summary>Bash</summary>
+```bash
+curl --insecure -u admin:admin -X POST -d '{"ownerX500Name": "O=Alice, L=London, C=GB"}' $API_URL/mgm/$MGM_HOLDING_ID/preauthtoken
+```
+</details>
+
+This REST API allows for additional optional properties to be submitted also when creating a token. The first is a time-to-live, which allows a duration to be submitted, after which the token will no longer be valid for use. This duration is submitted in the `ISO-8601` duration format (i.e. `PnDTnHnMn.nS`). For example, PT15M (15 minutes), P4D (4 days), P1DT2H2M (1 day 2 hours and 2 minutes). The duration submitted is added to the current time when the request to create the token is submitted to calculate the time at which the token is no longer valid. If no time-to-live value is submitted, the token will only expire after it is consumed or revoked. The second optional property that can be submitted is a remark. This is simply a user defined string which will be stored along with the token as additional information which can provide information about the token creation when the token is viewed.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
This REST API allows for additional optional properties to be submitted also when creating a token. The first is a time-to-live, which allows a duration to be submitted, after which the token will no longer be valid for use. This duration is submitted in the `ISO-8601` duration format (i.e. `PnDTnHnMn.nS`). For example, PT15M (15 minutes), P4D (4 days), P1DT2H2M (1 day 2 hours and 2 minutes). The duration submitted is added to the current time when the request to create the token is submitted to calculate the time after which the token is no longer valid. If no time-to-live value is submitted, the token will only expire after it is consumed or revoked. The second optional property that can be submitted is a remark. This is simply a user defined string which will be stored along with the token as additional information which can provide information about the token creation when the token is viewed. 
```

File: wiki/Manual-Approval.md [hunk 3]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+
+Corda has a set of REST APIs available for managing these pre-auth tokens. Through these APIs, tokens can be created, revoked, and viewed. When viewing a token, it is possible to see the token ID, the X.500 name of the member the token is assigned to, optionally a time and date when the token expires, the token status, and additional information provided by the MGM when creating or revoking the token.
+
+## Creating a pre-auth token
+
+Token creation is done through a REST API using the POST method. At a minimum, the X.500 name of the member the token is for must be provided. This token will be tied to that X.500 name and only a registering member with the same X.500 name will be able to consume that token.
+
+<details>
+<summary>Bash</summary>
+```bash
```
Comment (YashNabar) [inline wiki/Manual-Approval.md:117]: Does this render correctly for you? It doesn't show me a code-block. Same for some other new ones below

File: wiki/Manual-Approval.md [hunk 4]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+curl --insecure -u admin:admin -X POST -d '{"ownerX500Name": "O=Alice, L=London, C=GB", "ttl": "P7D", "remarks": "Member was verified offline on 01/02/2023 and does not need additional verification when joining the network."}' $API_URL/mgm/$MGM_HOLDING_ID/preauthtoken
+```
+</details>
+
+## Viewing tokens
+
+To view tokens that have been created, you need to use the pre-auth token GET API. This returns a list of all tokens that the MGM has created which have not been consumed, revoked, or automatically invalidated by Corda (for example, due to an expired TTL).
+
+If you wish to view tokens which are inactive (i.e. consumed, revoked, or auto-invalidated), you can set the query parameter `viewInactive` equal to true and pre-auth tokens which are available will be returned along with tokens with are consumed, revoked, or auto-invalidated. If this is set to false, only tokens with are active and ready to use are returned.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
If you wish to view tokens which are inactive (i.e. consumed, revoked, or auto-invalidated), you can set the query parameter `viewInactive` equal to true and pre-auth tokens which are available will be returned along with tokens with are consumed, revoked, or auto-invalidated. If this is set to false, only tokens which are active and ready to use are returned.
```

File: wiki/Manual-Approval.md [hunk 5]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+
+<details>
+<summary>Bash</summary>
+```bash
+curl --insecure -u admin:admin $API_URL'/mgm/'$MGM_HOLDING_ID'/preauthtoken?viewinactive=false'
+```
+</details>
+
+This endpoint accepts optional parameters to filter or expand the search results. The first filter is the X.500 name of the member who the token was issued for. This is passed in as a URL query parameters called `ownerX500Name`. The full URL encoded X.500 name should be passed in here to filter correctly. The second filter is token ID. If you know the ID of a specific token you want to look up then you can provide that to the API as the query parameters `preAuthTokenId`.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
This endpoint accepts optional parameters to filter or expand the search results. The first filter is the X.500 name of the member who the token was issued for. This is passed in as a URL query parameter called `ownerX500Name`. The full URL encoded X.500 name should be passed in here to filter correctly. The second filter is token ID. If you know the ID of a specific token you want to look up then you can provide that to the API as the query parameter `preAuthTokenId`.  
```

File: wiki/Manual-Approval.md [hunk 6]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+
+# Configure manual approval of pre-authenticated registrations
+
+Pre-auth tokens allow registrations to skip the standard set of registration rules configured, though the network operator might still want to review certain changes to the member's context even if a pre-auth token was submitted. This is possible through a set of APIs very similar to the previously described APIs for defining registration approval rules. 
+
+These pre-auth approval rules are applied to the registration requests containing a valid pre-auth token in the same way as done for standard approval rules applied to registrations without a valid pre-auth token.
+
+## Add a pre-auth group approval rule
+
+This API will add a group approval rule used specifically for the registrations containing a valid pre-auth token. The body of the request is the same as for standard approval rules. It contains the regular expression which is applies to all changed keys in the member context, and a label to describe the rule for informational purposes.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
This API will add a group approval rule used specifically for the registrations containing a valid pre-auth token. The body of the request is the same as for standard approval rules. It contains the regular expression which is applied to all changed keys in the member context, and a label to describe the rule for informational purposes.
```

File: wiki/Manual-Approval.md [hunk 7]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+
+```bash
+RULE_PARAMS='{"ruleParams":{"ruleRegex": "^corda.endpoints.*$", "ruleLabel": "Any change to P2P endpoints requires manual review."}}'
+curl --insecure -u admin:admin -d $RULE_PARAMS $API_URL/mgm/$MGM_HOLDING_ID/approval/rules/preauth
+```
+</details>
+
+## View current pre-auth group approval rules
+
+Viewing the current pre-auth approval rules is also similar to the method of viewing standard group approval rules. This is a GET endpoint which shows how the group is currently configured in terms of automatically approving registrations requests with a valid pre-auth token.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
Viewing the current pre-auth approval rules is also similar to the method of viewing standard group approval rules. This is a GET endpoint which shows how the group is currently configured in terms of automatically approving registration requests with a valid pre-auth token.
```

File: wiki/Manual-Approval.md [hunk 8]
```diff
@@ -100,3 +100,175 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+RULE_ID=<RULE ID>
+curl --insecure -u admin:admin -X DELETE $API_URL/mgm/$MGM_HOLDING_ID/approval/rules/preauth/$RULE_ID
+```
+
+</details>
+
+
+# Manual approval of pre-authenticated registrations
+
+Viewing, approving or declining paused registration requests with pre-auth tokens included is done using the same API as registrations without pre-auth tokens. Refer to the section "How to manually approve/decline requests" above to see how this is done.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md]:
```suggestion
Viewing, approving or declining paused registration requests with pre-auth tokens included is done using the same API as registrations without pre-auth tokens. Refer to the section [How to manually approve/decline requests](#how-to-manually-approvedecline-requests) above to see how this is done.
```

**Review by YashNabar — COMMENTED** [review]
Looks good, added some very minor comments

File: wiki/Manual-Approval.md (hunk 3 above)
Comment (charlieR3) [reply wiki/Manual-Approval.md:117]: Whoops! Forgot to check how it rendered. Will fix that. Thanks.

File: wiki/Manual-Approval.md [hunk 9]
```diff
@@ -100,3 +100,186 @@ curl --insecure -u admin:admin -d "$REASON" $API_URL/mgm/$MGM_HOLDING_ID/decline
...
+<details>
+<summary>Bash</summary>
+
+```bash
+curl --insecure -u admin:admin $API_URL'/mgm/'$MGM_HOLDING_ID'/preauthtoken'
+```
+
+</details>
+
+This endpoint accepts optional parameters to filter or expand the search results. The first filter is the X.500 name of the member who the token was issued for. This is passed in as a URL query parameter called `ownerX500Name`. The full URL encoded X.500 name should be passed in here to filter correctly. The second filter is token ID. If you know the ID of a specific token you want to look up then you can provide that to the API as the query parameter `preAuthTokenId`.  And the last filter is used to view tokens which are inactive (i.e. consumed, revoked, or auto-invalidated). To use this filter, set the query parameter `viewInactive` equal to true and pre-auth tokens which are available will be returned along with tokens with are consumed, revoked, or auto-invalidated. If this is set to false, only tokens which are active and ready to use are returned.
```
Comment (YashNabar) [inline wiki/Manual-Approval.md:147]: One last comment, very minor
```suggestion
This endpoint accepts optional parameters to filter or expand the search results. The first filter is the X.500 name of the member who the token was issued for. This is passed in as a URL query parameter called `ownerX500Name`. The full URL encoded X.500 name should be passed in here to filter correctly. The second filter is token ID. If you know the ID of a specific token you want to look up then you can provide that to the API as the query parameter `preAuthTokenId`.  And the last filter is used to view tokens which are inactive (i.e. consumed, revoked, or auto-invalidated). To use this filter, set the query parameter `viewInactive` equal to true and pre-auth tokens which are available will be returned along with tokens which are consumed, revoked, or auto-invalidated. If this is set to false, only tokens which are active and ready to use are returned.
```

**Review by YashNabar — APPROVED** [review]
LGTM

**Comment by charlieR3** [comment]
Discussed at stand up. Can be closed.

**Pull request closed without merging**